test/v4/controllers/createUkPropertyPeriodSummary/def2/Def2_CreateUkPropertyPeriodSummaryValidatorSpec.scala

```diff
@@ -0,0 +1,530 @@
+        setupMocks()
+        List(
+          "/ukFhlProperty/income/periodAmount",
+          "/ukFhlProperty/income/taxDeducted",
+          "/ukFhlProperty/income/rentARoom/rentsReceived",
+          "/ukFhlProperty/expenses/premisesRunningCosts",
+          "/ukFhlProperty/expenses/repairsAndMaintenance",
+          "/ukFhlProperty/expenses/financialCosts",
+          "/ukFhlProperty/expenses/professionalFees",
+          "/ukFhlProperty/expenses/costOfServices",
+          "/ukFhlProperty/expenses/other",
+          "/ukFhlProperty/expenses/travelCosts",
+          "/ukFhlProperty/expenses/rentARoom/amountClaimed",
+          "/ukNonFhlProperty/income/premiumsOfLeaseGrant",
+          "/ukNonFhlProperty/income/reversePremiums",
+          "/ukNonFhlProperty/income/periodAmount",
+          "/ukNonFhlProperty/income/taxDeducted",
+          "/ukNonFhlProperty/income/otherIncome",
+          "/ukNonFhlProperty/income/rentARoom/rentsReceived",
+          "/ukNonFhlProperty/expenses/premisesRunningCosts",
+          "/ukNonFhlProperty/expenses/repairsAndMaintenance",
+          "/ukNonFhlProperty/expenses/financialCosts",
+          "/ukNonFhlProperty/expenses/professionalFees",
+          "/ukNonFhlProperty/expenses/costOfServices",
+          "/ukNonFhlProperty/expenses/other",
+          "/ukNonFhlProperty/expenses/residentialFinancialCost",
+          "/ukNonFhlProperty/expenses/travelCosts",
+          "/ukNonFhlProperty/expenses/residentialFinancialCostsCarriedForward",
+          "/ukNonFhlProperty/expenses/rentARoom/amountClaimed"
+        ).foreach(path => testValueFormatErrorWith(path, validBody.update(path, JsNumber(123.456))))
+      }
+
+      "passed a body with a (consolidated expenses) field containing an invalid value" when {
+        setupMocks()
+        List(
+          "/ukFhlProperty/expenses/consolidatedExpenses",
+          "/ukNonFhlProperty/expenses/consolidatedExpenses"
+        ).foreach(path => testValueFormatErrorWith(path, validBodyConsolidatedWithExtraFields.update(path, JsNumber(123.456))))
+      }
+
+      "passed a body with multiple invalid fields" in {
+        setupMocks()
+        val path0 = "/ukFhlProperty/expenses/travelCosts"
+        val path1 = "/ukNonFhlProperty/expenses/travelCosts"
+        val path2 = "/ukNonFhlProperty/expenses/rentARoom/amountClaimed"
+
+        val invalidBody = validBody
+          .update(path0, JsNumber(123.456))
+          .update(path1, JsNumber(123.456))
+          .update(path2, JsNumber(123.456))
+
+        val result: Either[ErrorWrapper, CreateUkPropertyPeriodSummaryRequestData] =
+          validator(validNino, validTaxYear, validBusinessId, invalidBody).validateAndWrapResult()
+
+        result shouldBe Left(ErrorWrapper(correlationId, ValueFormatError.withPaths(List(path0, path1, path2))))
+      }
+
+      "passed a body with both consolidated and separate expenses provided for fhl" in {
```

Comment:
Please add tests where extra fields are supplied with validBodyConsolidatedWithExtraFields in both fhl and nonFhl bodies to check if RuleBothExpensesSuppliedError is returned in those scenarios.